pkg/volume/util/util.go

```diff
@@ -571,6 +571,34 @@ func IsLocalEphemeralVolume(volume v1.Volume) bool {
 		volume.ConfigMap != nil
 }
 
+// GetLocalLocalPersistentVolumeNodeName determines whether the argument is a local PersistentVolume
+// and return the node name for the PV node affinity if it is a local PV.
+// Local PV has node affinity in the form of:
+// nodeAffinity:
+//	required:
+//	  nodeSelectorTerms:
+//	  - matchExpressions:
+//		- key: kubernetes.io/hostname
+//		  operator: In
+//		  values:
+//		  - <node>
+func GetLocalLocalPersistentVolumeNodeName(pv *v1.PersistentVolume) (string, bool) {
+	if pv != nil && pv.Spec.NodeAffinity != nil && pv.Spec.NodeAffinity.Required != nil {
+		if len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) == 1 {
+			nodeSelector := pv.Spec.NodeAffinity.Required.NodeSelectorTerms[0]
+			if len(nodeSelector.MatchExpressions) == 1 {
+				matchExpr := nodeSelector.MatchExpressions[0]
+				if matchExpr.Key == v1.LabelHostname && matchExpr.Operator == v1.NodeSelectorOpIn {
+					if len(matchExpr.Values) == 1 {
+						return matchExpr.Values[0], true
+					}
```

Comment:
I'm thinking of relax the value check, so basically returns all legal host names instead of a single one? (and probably also relax the `len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms)` check)

@cofyc could you confirm this is the most common pattern to specify local PV?

Reply:
AFAIK, local PV that is using the static provisioner have PV node affinity created like:
https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/pkg/discovery/discovery.go#L198

which is a single match expressions for key `kubernetes.io/hostname` with operation `In` and value the node name.

@msau42 @jsafrane can chime in if needed. Will bring this up with sig-storage.

Reply:
as for local PVs provisioned by local-static-provisioner, there is only one hostname bound to PV

> I'm thinking of relax the value check, so basically returns all legal host names instead of a single one?

but +1 for this in scheduling, we can add all values under `kubernetes.io/hostname` key in potential nodes